### DISCUSSION TITLE
How long does make package take to finish?

### DISCUSSION THREAD

**Discussion by varunjammula** [carla-simulator/carla#4973] [Q&A] [answered]
Hi, I am building CARLA 0.9.13 on Linux(Ubuntu-18). make launch and make PythonAPI worked well without any errors. I am trying to package the whole thing and running the make package. The build has been running for more than 3hrs. Did anyone do this before and how long is it supposed to take? I am currently stuck at LogShaderCompilers: Display: Worker (3/9): shaders left to compile 1

![image](https://user-images.githubusercontent.com/3236409/145767298-d2ea7cd5-15b7-4572-b213-2301fdfa2158.png)

**Comment by ynlch** [comment, marked as answer]
I run make package for carla 0.9.12 about 27 minutes
real	27m30.506s
user	14m12.204s
sys	1m43.509s

**Reply by varunjammula** [reply]
Thanks for the reply. My build succeeded as well but took more time.

**Reply by xiaoqingnianbudadun** [reply]
i have a question, how can you get the carla branch of UE4? I can't get the branch in UE4, so i can not build success.

**Reply by varunjammula** [reply]
https://carla.readthedocs.io/en/latest/build_linux/#unreal-engine

**Reply by xiaoqingnianbudadun** [reply]
i have found th carla branch of UE4 here :   https://github.com/CarlaUnreal/UnrealEngine.git
